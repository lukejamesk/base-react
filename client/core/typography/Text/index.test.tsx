import React from 'react'
import { mount } from 'enzyme'
import { Text } from '.'
import { forEach } from 'ramda'

const textTypes = ['p', 'strong', 'blockquote']

describe('<Text />', () => {
  forEach(type => {
    it(`should render with: ${type}`, () => {
      const wrapper = mount(<Text type={type}>My content for {type}</Text>)
      expect(wrapper).toMatchSnapshot()
    })
  }, textTypes)
})
