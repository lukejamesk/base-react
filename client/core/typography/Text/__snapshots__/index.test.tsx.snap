// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render with: blockquote 1`] = `
<ForwardRef
  type="blockquote"
>
  <blockquote>
    My content for
    blockquote
  </blockquote>
</ForwardRef>
`;

exports[`<Text /> should render with: p 1`] = `
<ForwardRef
  type="p"
>
  <p>
    My content for
    p
  </p>
</ForwardRef>
`;

exports[`<Text /> should render with: strong 1`] = `
<ForwardRef
  type="strong"
>
  <strong>
    My content for
    strong
  </strong>
</ForwardRef>
`;
